tests/_test_main_helpers.py

```
@@ -0,0 +1,474 @@
+Extracted from tests/test_main.py to keep test modules under 2000 lines.
+"""
+
+import asyncio
+import concurrent.futures
+import contextlib
+import functools
+import inspect
+import sys
+import time
+from concurrent.futures import Future
+from typing import Any, Callable
```

Ruff (0.15.12)

[warning] Import from `collections.abc` instead: `Callable`

Import from `collections.abc`

(UP035)